opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -352,22 +360,43 @@ private ExprValue parseStruct(Content content, String prefix) {
    * <a href="https://opensearch.org/docs/latest/opensearch/supported-field-types/nested/">docs</a>
    * The similar data type is nested, but it can only allow a list of objects.
    */
-  private ExprValue parseArray(Content content, String prefix) {
+  public ExprValue parseArray(Content content, String prefix) {
     List<ExprValue> result = new ArrayList<>();
     // ExprCoreType.ARRAY does not indicate inner elements type.
-    if (Iterators.size(content.array()) == 1 && content.objectValue() instanceof JsonNode) {
-      result.add(parse(content, prefix, Optional.of(STRUCT)));
-    } else {
-      content.array().forEachRemaining(v -> {
-        // ExprCoreType.ARRAY does not indicate inner elements type. OpenSearch nested will be an
-        // array of structs, otherwise parseArray currently only supports array of strings.
-        if (v.isString()) {
-          result.add(parse(v, prefix, Optional.of(OpenSearchDataType.of(STRING))));
-        } else {
-          result.add(parse(v, prefix, Optional.of(STRUCT)));
+
+    try {
```

Comment:
As of right now, I'm using the try/catch statment to determine if the content value is an array.  I'm not a huge fan of this implementation, and there is an alternative solution which would involve adding a new `getValue` method that would get the raw `JsonElement value` instance variable from the `Content` object.  However, this method would need to be added to the `Content`, `OpenSearchJsonContent`, and `ObjectContent` classes even though it only applies to the JSON logic.  I would appreciate some input on which solution that we should move forward with.

Reply:
This method reworked in upcoming changes in https://github.com/Bit-Quill/opensearch-project-sql/pull/274. Probably this can give you an idea what to do.

Reply:
That PR addresses most of the issues I was running into while attempting to fix the integration tests.  Thanks!